typing/env.ml

```diff
@@ -471,7 +472,7 @@ module IdTbl =
         | Nothing -> raise exn
         end
 
-    let find_same id (tbl : (empty, _, _) t) =
+    let find_same id (tbl : (_, _, _) t) =
```

Comment:
The diff makes the code worse. (we used to ensure that the locks must be of type `empty` and thus safe to ignore)